src/mip/coefficient_strengthening.jl

```diff
@@ -0,0 +1,178 @@
+    new_bound = ps.urow[i]
+    new_coef = coef
+    if coef > 0
+        d = new_bound - max_act - coef * (ps.ucol[j]-1)
+        if coef >= d > 0
+            new_coef = new_coef - d
+            new_bound -= d*ps.ucol[j]
+        end
+    elseif coef < 0
+        d = new_bound - max_act - coef * (ps.lcol[j]+1)
+        if -coef >= d > 0
+            new_coef = new_coef + d
+            new_bound += d*ps.lcol[j]
+        end
+    end
+    return new_coef, new_bound
+end
+
+function lowerbound_strengthening(ps::PresolveData{T}, i::Int, j::Int, coef, min_act::T) where {T}
+    # perform coef strengthening for one constraint of the form l < = a'x
+    new_bound = ps.lrow[i]
+    new_coef = coef
+    if coef > 0
+        d = -new_bound + min_act + coef*(ps.lcol[j]+1)
+        if coef >= d > 0
+            new_coef = coef - d
+            new_bound -= d*ps.lcol[j]
+        end
+    elseif coef < 0
+        d = -new_bound + min_act + coef*(ps.ucol[j]-1)
+        if -coef >= d > 0
+            new_coef = coef + d
+            new_bound += d*ps.ucol[j]
+        end
+    end
+    return new_coef, new_bound
+end
+
+function zero_coefficient_strengthening!(ps::PresolveData{T}) where {T}
+    # perform coefficient stregthening but if there is a coefficient that is reduced to 0
+    # it is explicitly stored in the ps.pb0.arows and ps.pb0.acols
+
+    # keep track of index for each var fo update ps.acols
+    # use this to find which index of ps.pb0.acols[i].nzval to update
+    i_index = zeros(Int, ps.pb0.nvar)
+
+    for i in 1:ps.pb0.ncon
+        ps.rowflag[i] || continue
+
+        lrow = ps.lrow[i]
+        urow = ps.urow[i]
+        if isfinite(lrow) && isfinite(urow) #skip ranged constraints
+            continue
+        end
+
+        row = ps.pb0.arows[i]
+        if all(ps.var_types[row.nzind] .== CONTINUOUS) # at least 1 integer
+            continue
+        end
+
+        sup = maximal_activity(ps, i)
+        inf = minimal_activity(ps, i)
+
+        j_index = 0 # keep track of index of variable j in row.nzind & row.nzval
+        for j in row.nzind
+            j_index += 1
+            i_index[j] += 1
+            if ps.var_types[j] == CONTINUOUS || !ps.colflag[j]
+                continue
+            else
+                coef = row.nzval[j_index]
+
+                if isfinite(urow)
+                    if coef > 0
+                        max_act = sup - coef * ps.ucol[j] # maximal activity of every variables except j
+                    else
+                        max_act = sup - coef * ps.lcol[j]
+                    end
+                    new_coef, new_bound = upperbound_strengthening(ps, i, j, coef, max_act)
+                    # update problem
+                    row.nzval[j_index] = new_coef
+                    ps.pb0.acols[j].nzval[i_index[j]] = new_coef
+                    ps.urow[i] = new_bound
+                    #update nonzero
+                    if new_coef == 0
```

Comment:
Last thing I could spot: the number of row/column non-zeros should be decremented only if a non-zero coefficient is set to zero: if `coef == new_coef == 0`, the number of non-zeros should remain the same.
```suggestion
                    if !iszero(coef) && iszero(new_coef)
```